.github/workflows/publish.yml

```diff
@@ -2,60 +2,56 @@ name: Publish
 
 on:
   push:
-    branches: ["main"]
-    tags: ["*.*.*"]
-    paths-ignore: ["*.md"]
+    branches:
+    - main
 
-env:
-  CI: true
-
-# Pnpm setup based on https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time
+concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
-  version:
+  release:
+    name: Release
     runs-on: ubuntu-latest
     permissions:
       id-token: write # required for provenance https://docs.npmjs.com/generating-provenance-statements#publishing-packages-with-provenance-via-github-actions
       contents: write
       pull-requests: write
 
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
+      - uses: actions/checkout@v4
 
       - name: Install Node.js
         uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache: "pnpm"
 
       - name: Install pnpm
         uses: pnpm/action-setup@v4
-        id: pnpm-install
         with:
           version: 9
           run_install: false
 
       - name: Get pnpm store directory
-        id: pnpm-cache
         shell: bash
         run: |
-          echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
```

Comment:
PNPM uses `$GITHUB_ENV`. `$GITHUB_ENV` only affects the current steps, not other steps that may be part of this job.